### DISCUSSION TITLE
Is it possible to have an `Event Trigger` on a `Hasura DB` to fire only `once` when a target table is `updated` / `data inserted` ?

### DISCUSSION THREAD

**Discussion by migbash** [hasura/graphql-engine#9266] [Question] [answered]
In my team we have the `hasura` data being also further cached and requires events to fire **once** when table has been completely updated after a `target` query completion.

Currently, when I select for an `Event Trigger` to `fire` on an event `update` or `insert` for a target table, the events fire **X** amount of times, for where **X** is the number of `affected` rows of the table.

<img width="213" alt="image" src="https://user-images.githubusercontent.com/20924663/205710757-e86fd8a2-8207-4662-b134-8b7e67dc0a98.png">

Is it possible to `fire` only **ONCE** the `Event Trigger` when our backend has executed a target `Upsert` / `Insert` query ? 

Otherwise, for example, if 150 rows where affected by the `Upsert` query, 150 individual `Event Triggers` are fired, causing an overload - not ideal.

**Comment by adas98012** [comment, marked as answer]
@migbash 
Hasura relies on postgres database triggers (in the backend) for event triggers and specifically creates triggers for EACH ROW.
It is not possible to have one Event Trigger for a batch INSERT/UPDATE, in the current product.

However, you can try a workaround solution which involves the following steps : 

-  Create a new table (to be used as a proxy)

-  Create a postgres trigger that inserts records into the proxy table (after a bulk insert/update on the target tables). The trigger must be executed for EACH STATEMENT (not EACH ROW)  [Doc](https://www.postgresql.org/docs/current/sql-createtrigger.html)

- Create a hasura event trigger based on inserts to the proxy table
